cogeo_mosaic/backends/file.py

```diff
@@ -0,0 +1,66 @@
+"""cogeo-mosaic File backend."""
+
+from typing import Dict, List, Optional, Union
+
+import json
+import functools
+
+import mercantile
+
+from cogeo_mosaic.model import MosaicJSON
+from cogeo_mosaic.backends.base import BaseBackend
+from cogeo_mosaic.backends.utils import (
+    _compress_gz_json,
+    _decompress_gz,
+    get_assets_from_json,
+)
+
+
+class FileBackend(BaseBackend):
+    """Local File Backend Adapter"""
+
+    def __init__(self, path: str, mosaic_def: Optional[Union[MosaicJSON, Dict]] = None):
+        """Initialize FileBackend."""
+        self.path = path
+
+        if mosaic_def is not None:
+            self.mosaic_def = MosaicJSON(**dict(mosaic_def))
+        else:
+            self.mosaic_def = self.read()
+
+    def tile(self, x: int, y: int, z: int) -> List[str]:
+        """Retrieve assets for tile."""
+        return get_assets_from_json(self.mosaic_def.tiles, self.quadkey_zoom, x, y, z)
+
+    def point(self, lng: float, lat: float) -> List[str]:
+        """Retrieve assets for point."""
+        tile = mercantile.tile(lng, lat, self.quadkey_zoom)
+        return get_assets_from_json(
+            self.mosaic_def.tiles, self.quadkey_zoom, tile.x, tile.y, tile.z
+        )
+
+    def write(self, gzip=None):
+        """Write mosaicjson document to a file."""
+        body = dict(self.mosaic_def)
+        if gzip or (gzip is None and self.path.endswith(".gz")):
+            body = _compress_gz_json(body)
+        else:
+            body = json.dumps(body).encode("utf-8")
+
+        with open(self.path, "wb") as f:
```

Comment:
Should we allow to save mosaic in non-binary format ?

Reply:
Well a "text format" is binary encoded as UTF-8, so I don't see a difference.